src/main/java/Duke.java

```diff
@@ -1,10 +1,81 @@
+import exception.DukeException;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Represents the attributes of the chatbot Duke. Every instantiation of a new Duke object
+ * allows features such as adding, deleting, listing, loading, saving, marking and unmarking of
+ * different types of tasks into a task list.
+ */
 public class Duke {
+    protected TaskList taskList;
+    protected FileDataHandler fileDataHandler;
+    protected Ui ui;
+
+    /**
+     * Creates new TaskList, Ui and FileDataHandler objects given the filePath and name of directory.
+     * Creates and loads the saved file data, if the file does not exist, throws an error message.
+     *
+     * @param filePath the path to the .txt file where the saved data will be stored.
+     * @param directoryName The name of the folder where the saved file is to be located.
+     */
+    public Duke(String filePath, String directoryName) {
+        taskList = new TaskList();
+        fileDataHandler = new FileDataHandler(filePath, directoryName);
+        ui  = new Ui();
+        try {
+            if(!fileDataHandler.createFile()) {
+                fileDataHandler.loadFile(taskList);
+            }
+        } catch (FileNotFoundException exception) {
+            ui.printErrorMessage("     ☹ OOPS!!! The save file could not be loaded as it does not exist.");
+        } catch (IOException exception) {
+            ui.printErrorMessage("     ☹ OOPS!!! As an error has occurred, the current task list will not be saved in a file!");
```

Comment:
Perhaps avoid magic literals, consider storing the error messages in constants (static final vars). [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)